WooCommerce/Classes/POS/ViewModels/PointOfSaleDashboardViewModel.swift

```diff
@@ -208,46 +265,55 @@ private extension PointOfSaleDashboardViewModel {
             }
         }.assign(to: &$showsCardReaderSheet)
     }
-}
 
-import enum Yosemite.OrderAction
-import struct Yosemite.Order
-@MainActor
-private extension PointOfSaleDashboardViewModel {
-    private func createOrUpdateTestOrder() async throws -> Order {
-        if let order {
-            let updatedOrder = try await update(order: order, amount: formattedOrderTotalPrice ?? "15.00")
-            self.order = updatedOrder
-            return updatedOrder
-        } else {
-            let order = try await createTestOrder(amount: formattedOrderTotalPrice ?? "15.00")
+    @MainActor
+    private func syncOrder(for cartProducts: [CartItem]) async {
+        guard isSyncingOrder == false else {
+            return
+        }
+        isSyncingOrder = true
+        let cart = cartProducts
+            .map {
+                POSCartItem(itemID: nil,
+                            product: $0.item,
+                            quantity: Decimal($0.quantity))
+            }
+        defer {
+            self.isSyncingOrder = false
+        }
+        do {
+            self.isSyncingOrder = true
+            let order = try await self.orderService.syncOrder(cart: cart,
+                                                              order: self.order,
+                                                              allProducts: self.items)
```

Comment:
```suggestion
            isSyncingOrder = false
        }
        do {
            isSyncingOrder = true
            let order = try await orderService.syncOrder(cart: cart,
                                                         order: order,
                                                         allProducts: items)
```

No need for `self` here